packages/core/src/providers/netsuite.ts

```diff
@@ -0,0 +1,238 @@
+ *           }
+ *         },
+ *         // Optional
+ *         prompt: "login", // Required if you want to force the user to login every time.
+ *         scope: "restlets", // Optional defaults to "restlets rest_webservices". Enter the scope(s) you want to use followed by spaces.
+ *       })
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *
+ * - [NetSuite - Creating an Integration Record (OAuth 2.0)](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771733782.html#Related-Topics)
+ * - [NetSuite - Authorizing OAuth Requests](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps)
+ * - [NetSuite - Configure OAuth Roles](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771510070.html#Set-Up-OAuth-2.0-Roles)
+ * - [Learn more about NetSuite OAuth 2.0](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/chapter_157769826287.html#OAuth-2.0)
+ * - [Source code](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/netsuite.ts)
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the NetSuite provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ * Make sure the `userInfoUrl` matches the return type of the profile callback to ensure the user session gets read correctly.
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/providers/custom-provider#override-default-options).
+ *
+ * :::
+ *
+ * :::info **Disclaimer**
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function NetSuite(
+  config: OAuthNetSuiteOptions
+): string | object {
+  return {
+    id: "netsuite",
+    name: "NetSuite",
+    type: "oauth",
+    version: "2.0",
+    protection: 'state',
+    headers: {redirect: 'follow'},
+    checks: ["state"],
+    authorization: {
+      url: `https://${config.accountId}.app.netsuite.com/app/login/oauth2/authorize.nl`,
+      params: {
+        client_id: `${config.clientId}`,
+        prompt: config?.prompt ?? 'login',
+        response_type: 'code',
+        scope: config.scope || 'restlets rest_webservices',
+      }
+    },
+    token: {
+      url: `https://${config.accountId}.suitetalk.api.netsuite.com/services/rest/auth/oauth2/v1/token`,
+      params: {
+        grant_type: "authorization_code",
+      }
+    },
+    userinfo: `${config.userInfoUrl}`,
+    profile: (profile: Object, token) => config.profileCallback(profile, token),

```

Comment:
Ah sorry, I thought we'd changed this already. We shouldn't require the user to pass a profile callback. Lets use the default here

```suggestion
-    profile: (profile: Object, token) => config.profileCallback(profile, token),
+    profile(profile) {
+      return {
+        id: profile.id.toString(),
+        name: profile.name ?? profile.username,
+        email: profile.email,
+        image: profile.avatar_url,
+      }
+    },
```

And then folks can always still use the `profile` option of the provider to override anything if they want to add more fields, etc.

```ts
  providers: [
    NetSuite({
      profile(profile) {
        return { role: profile.role ?? "user", ... }
      },
    })
  ],
```

Reply:
I have implemented new defaults @ndom91 along with new documentation. Let me know if this is in anyway confusing. Lol I have also changed some props on the options to mimic the Salesforce provider.